lib/daru/category.rb

```diff
@@ -922,6 +922,13 @@ def modify_category_at pos, category
       @cat_hash[category] << pos
     end
 
+    def set indexes, val
+      @index = @index.add(indexes)
+      @cat_hash.store(val,[indexes])
+      @array << indexes
+      modify_category_at indexes, val
+    end
+
```

Comment:
`@array` is supposed to store categories (in integer form) not indexes. That's what you are are getting error here.

Reply:
I doubt that because ` puts @array` gives `0,1,2,5`.

Reply:
0, 1, 2 refers to categories not indexes in @array